localboot/grub.go

```diff
@@ -95,35 +124,53 @@ func ParseGrubCfg(grubcfg string, basedir string, grubVersion int) []bootconfig.
 	return bootconfigs
 }
 
+func unquote(ver grubVersion, text string) string {
+	if ver == grubV2 {
+		// if grub2, unquote the string, as directives could be quoted
+		// https://www.gnu.org/software/grub/manual/grub/grub.html#Quoting
+		// TODO unquote everything, not just \$
+		return strings.Replace(text, `\$`, "$", -1)
+	}
+	// otherwise return the unmodified string
+	return text
+}
+
 // ScanGrubConfigs looks for grub2 and grub legacy config files in the known
 // locations and returns a list of boot configurations.
 func ScanGrubConfigs(basedir string) []bootconfig.BootConfig {
 	bootconfigs := make([]bootconfig.BootConfig, 0)
-	// Scan Grub 2 configurations
-	for _, grubpath := range Grub2Paths {
-		path := path.Join(basedir, grubpath)
-		log.Printf("Trying to read %s", path)
-		grubcfg, err := ioutil.ReadFile(path)
-		if err != nil {
-			log.Printf("cannot open %s: %v", path, err)
-			continue
+	err := filepath.Walk(basedir, func(currentPath string, info os.FileInfo, err error) error {
```

Comment:
`filepath.Walk` recurses the whole subtree, this is not a good idea because it may be very large. I would rather stick with a list of known paths

Reply:
discussed with @zaolin in person. We should find a middle ground here. Recursing this way is not a good idea, and Grub/Grub2 don't make this task easy - it's completely out of control where config files could be located. We can only make guesses unless we want implement a full-fledged parser (spoiler: no, we don't want).

However if we recursively walk a file tree, we need safeguards.

Also CC @marcoguerri

Reply:
Right. I've added a limitation to the directory depth.